### PR TITLE
Fix #6462 Backend: Add support for filter by 'teamType' in teams API

### LINKED ISSUE #6462: need API support for filter by 'teamType' in teams API

### PR DESCRIPTION
**Pull request by vivekratnavel**
### Describe your changes :
See #6462 

 - Now `parentTeam` filter is added to List Teams API. So, we can get a list of all the teams under an organization with `/api/v1/teams?parentTeam=organization&fields=userCount,childrenCount`
 
#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
Screen Shot 2022-08-02 at 10 27 40 PM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java [hunk 1]
```diff
@@ -1814,6 +1815,86 @@ default Class<Team> getEntityClass() {
...
     }
+
+    @Override
+    default int listCount(ListFilter filter) {
+      String parentTeam = filter.getQueryParam("parentTeam");
+      String condition = filter.getCondition();
+      if (parentTeam != null) {
+        // validate parent team
+        Team team = EntityDAO.super.findEntityByName(parentTeam);
+        condition =
```
Comment (sureshms) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java:1826]: you can just call findEntityByName here right?

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TeamRepository.java [hunk 2]
```diff
@@ -165,6 +167,16 @@ private List<EntityReference> getUsers(Team team) throws IOException {
...
+  private Integer getUserCount(UUID teamId) throws IOException {
+    List<EntityRelationshipRecord> userIds = findTo(teamId, TEAM, Relationship.HAS, Entity.USER);
+    int userCount = userIds.size();
+    List<EntityReference> children = getChildren(teamId);
+    for (EntityReference child : children) {
+      userCount += getUserCount(child.getId());
+    }
+    return userCount;
+  }
+
```
Comment (sureshms) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TeamRepository.java:179]: For a team are we giving the count of all it's children and children's children? Just giving immediate children count is not sufficient?

File: catalog-rest-service/src/main/resources/json/schema/entity/teams/team.json [hunk 3]
```diff
@@ -76,6 +76,14 @@
       "$ref": "../../type/entityReference.json#/definitions/entityReferenceList",
       "default": null
     },
+    "childrenCount": {
+      "description" : "Total count of Children teams. An `Organization` can have `BusinessUnit`, `Division` or `Department` as children. A `BusinessUnit` can have `BusinessUnit`, `Division`, or `Department` as children. A `Division` can have `Division` or `Department` as children. A `Department` can have can have `Department` as children.",
+      "type": "integer"
```
Comment (sureshms) [inline catalog-rest-service/src/main/resources/json/schema/entity/teams/team.json:81]: We can just leave this as childrenCount description. I think what each type of team can have as children is captured as part of children property. Is this count only immediate children or all the children?

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TeamRepository.java (hunk 2 above)
Comment (harshach) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TeamRepository.java:179]: I think this is required show at the landing page of teams how many users are in a parent team. Example Cloud Infra as a department can have multiple teams under it but when a user is looking from org level we can show how many total users under it and how many total teams under it

**Comment by vivekratnavel** [comment]
@sureshms @harshach I have taken care of backward compatibility as well. Please review.

**Review by harshach — APPROVED** [review]

**Pull request merged**